### PR TITLE
Split images and add CI script to run e2e in-cluster

### PR DESCRIPTION
**Pull request by tnozicka**
This PR:
- splits the runtime and e2e tests image
- embeds kubectl into the e2e tests image where it is required (and was previously; this removes the mount hack)
- adds ci script to run e2e's on remote cluster (for Prow)

```[tasklist]
# Requires
- [ ] #33
- [ ] #35 
- [ ] #37 
```

### CODE REVIEW

**Comment by tnozicka** [comment]
@zimnx this is ready to be reviewed.
(Let Tide merge it when you're tagging it. I'll switch the required jobs on the repo when it has the tags.)

File: hack/.ci/run-e2e-gke.sh [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  gather-artifacts || "Error gathering artifacts" > /dev/stderr
+}
+
+trap handle-exit EXIT
+
+# Pre-create e2e namespace to be available to artifacts collection if something we to go wrong while deploying the stack.
+kubectl create namespace e2e --dry-run=client -o yaml | kubectl_create -f -
+kubectl create clusterrolebinding e2e --clusterrole=cluster-admin --serviceaccount=e2e:default --dry-run=client -o yaml | kubectl_create -f -
+
+timeout -v 10m ./hack/ci-deploy.sh "${E2E_IMAGE}"
```
[the remark this answers is not in the document]
Comment (tnozicka) [reply hack/.ci/run-e2e-gke.sh:83]: I kind of wanted it to be rather adjusted on demand + the way it was done was racy a bit but I can bring it back.

[the remark this answers is not in the document]
Comment (tnozicka) [reply hack/.ci/run-e2e-gke.sh:83]: added

**Comment by zimnx** [comment]
Thanks!
/lgtm

**Pull request merged**